src/pages/cart-drawer/drawer-cart-box-item/drawer-cart-box-item-control.tsx

```diff
@@ -0,0 +1,86 @@
+import { IconButton } from "@mui/material";
+import { Dispatch, FC, useEffect } from "react";
+import {
+  ControlBox,
+  CountBox,
+  TypographyQuantity,
+} from "./drawer-cart-box-item-styled";
+import DeleteOutlinedIcon from "@mui/icons-material/DeleteOutlined";
+import RemoveOutlinedIcon from "@mui/icons-material/RemoveOutlined";
+import AddOutlinedIcon from "@mui/icons-material/AddOutlined";
+import { CartProductModel } from "../../../api/api-cart/api-get-cart";
+import { useCartContext } from "../../../context/use-cart-context";
+
+const iconSize: string = "25px !important";
+
+interface DrawerCartBoxItemControlInterface {
+  setCartItemQuantity: Dispatch<React.SetStateAction<number>>;
+  cartItemQuantity: number;
+  cartItem: CartProductModel;
+}
+
+const DrawerCartBoxItemControl: FC<DrawerCartBoxItemControlInterface> = ({
+  cartItemQuantity,
+  setCartItemQuantity,
+  cartItem,
+}) => {
+  const cartContext = useCartContext();
+
+  const countCartProduct = (action: string) => {
+    cartContext.setHandleButton(!cartContext.handleButton);
+    if (action != "addition") {
+      return setCartItemQuantity(cartItemQuantity - 1);
+    }
+    return setCartItemQuantity(cartItemQuantity + 1);
+  };
+
+  useEffect(() => {
+    cartContext.updateProductToCart({
+      productUuid: cartItem.product.uid,
+      quantity: cartItemQuantity,
+    });
+  }, [cartItemQuantity, cartContext.handleButton, cartContext, cartItem.product.uid]);
+
+  return (
+    <ControlBox>
+      <IconButton
+        onClick={() => cartContext.deleteProductInCart(cartItem.product.uid)}
+      >
+        <DeleteOutlinedIcon sx={{ width: iconSize, height: iconSize }} />
+      </IconButton>
+
+      <CountBox>
+        <IconButton
+          disabled={
+            cartItemQuantity == 1 || cartContext.handleButton == true
+              ? true
+              : false
+          }
+          onClick={() => {
+            countCartProduct("subtraction");
+            cartContext.setHandleButton(!cartContext.handleButton);
+          }}
+        >
+          <RemoveOutlinedIcon />
+        </IconButton>
+        <TypographyQuantity>{cartItemQuantity}</TypographyQuantity>
+        <IconButton
+          onClick={() => {
+            countCartProduct("addition");
+            cartContext.setHandleButton(!cartContext.handleButton);
+          }}
+          disabled={
+            cartItemQuantity == cartItem.product.quantity ||
+            cartContext.handleButton == true
```

Comment:
you can do `cartContext.handleButton` no need to check if it is true just to return true

Reply:
resolve